src/mpid/ch3/src/mpid_finalize.c

```diff
@@ -148,6 +148,8 @@ int MPID_Finalize(void)
     MPIDI_RMA_finalize();
     
     MPL_free(MPIDI_failed_procs_string);
+    MPL_free(MPIR_Process.node_map);
+    MPL_free(MPIR_Process.node_local_map);
```

Comment:
Both of these are freed by `MPIR_pmi_finalize()`.  `MPIR_pmi_init/finalize` is being added to `ch3` in PR #4036: https://github.com/pmodels/mpich/pull/4036/commits/bae0a67f7282da844e8a05b4a74733239ab1561c

Reply:
I think I looked into integrating `MPIR_pmi_init/finalize` in CH3 too but for some reason I can't remember I gave up. I will look into it again and see if I can remember what was wrong.